tangent/grads.py

```diff
@@ -310,6 +310,17 @@ def adet(z, x):
   """
   adjugate = np.linalg.det(x) * np.linalg.pinv(x)
   d[x] = adjugate.T
+  
+
+@adjoint(numpy.linalg.inv)
+def ainv(z, x):
```

Comment:
It would be great to add some tests for these. To do that, you just need to add something in these lines to tests/functions.py:

    def inv(sqm):
      return np.linalg.inv(sqm)

That will trigger tests that compare the gradient with Autograd's (see #37 for some of the plumbing).

Note that the tests for 2.7 are currently broken, but 3.* should pass.

Reply:
I think this is done. I think I added the tests to this patch-3.  I don't understand what you mean by the correct adjoint, or d[z]. My understanding of how tangent works is still incomplete. I though z was just the inverse of the matrix in this case?

Reply:
Great, thanks!

Sorry if I was unclear, the mechanism is indeed a bit confusing. For an example, have a look at `@adjoint(gast.USub)`. The derivative of y = -x is y' = -1, but the respective adjoint is defined as d[x] = -d[y], accounting for the seed gradient in d[x] (and I think, but am not sure, that it follows from dy/dx = -1). This ensures the operations are chained properly. I'm just wondering how should d[z] be plugged into the equation in this case.

@alexbw, @bartvm, any thoughts?

Now that we have tests, we can verify it numerically, and since they are failing (see https://travis-ci.org/google/tangent/jobs/300286613#L583), the wiring is probably missing something.